### PR TITLE
Week 4

### PR DESCRIPTION
**Pull request by deepika845**
Added react Integration to the project.

### CODE REVIEW

File: my-app-react/src/App.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    for (let i = 0; i < currCartItems.length; i++) {
+      if (currCartItems[i].dishName === myname) {
+        updatedCartItems = [...currCartItems];
+        updatedCartItems[i].qty -= 1;
+        if (updatedCartItems[i].qty === 0) {
+          updatedCartItems = updatedCartItems.filter((curr) => {
+            return curr.dishName !== myname;
+          });
+        }
+        console.log("updated one : ", updatedCartItems);
```
Comment (smtrd3) [inline my-app-react/src/App.js]: @deepika845 please remove console.log

File: my-app-react/src/App.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    console.log("my menu items", activeMenuItems);
+  }, [activeMenu, currCartItems, activeMenuItems]);
+
+  return (
+    <React.StrictMode>
+      <Header />
+      <BreadCrumb />
+      <MiddleContainer>
+        <img src={FoodOffer} className="container-image"></img>
+
```
Comment (smtrd3) [inline my-app-react/src/App.js:118]: un-needed blank line, please integrate eslint to fix these style issue

File: my-app-react/src/Components/MenuList/MenuList.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+function MenuList(props) {
+
+  const selectTheMenuItem = (event) => {
+    console.log(event.target.textContent)
+    props.onSelect(event.target.textContent);
+    
+  };
+  // const [activeState, setActiveState] = useState('Recommended');
+  // function selectTheMenuItem(index) {
```
Comment (smtrd3) [inline my-app-react/src/Components/MenuList/MenuList.js]: in final PR there should not be any unnecessary comments.

File: script.js [hunk 4]
```diff
@@ -0,0 +1,429 @@
+import menuItems from "./Models/DishModel.js";
```
Comment (smtrd3) [inline script.js]: script.js name is not self explanatory, please use better name

File: script.js [hunk 5]
```diff
@@ -0,0 +1,429 @@
...
+      render();
+    }
+  }
+
+  function increaseCount(e) {
+    if (e.target.classList[0] === "in-menu--plus") {
+      const targetItem = e.target.parentNode.parentNode.parentNode;
+      const contentList = document.querySelector(".product-list");
+
+      /*<div class="selected-item-quantity">
```
Comment (smtrd3) [inline script.js]: remove comments

File: script.js [hunk 6]
```diff
@@ -0,0 +1,429 @@
...
+            <div class="add-remove-1">+</div>
+            <div class="add-remove-1">${qty}</div>
+            <div class="add-remove-1">-</div> */
+
+      const index = getIndexInMenu(targetItem, contentList);
+      console.log(index + "hikee");
```
Comment (smtrd3) [inline script.js]: no console.logs

File: script.js [hunk 7]
```diff
@@ -0,0 +1,429 @@
...
+
+      const { isVeg, dishName, price } = state.activeMenuList[index];
+
+      for (let i = 0; i < state.cart.length; i++) {
```
Comment (smtrd3) [inline script.js]: try using map/filter/reduce/forEach instead of plain for loop

File: script.js [hunk 8]
```diff
@@ -0,0 +1,429 @@
...
+      state = updatedState;
+    }
+    render();
+  }
+  function decreaseCount(e) {
+    if (e.target.classList[0] === "in-menu--minus") {
+      const targetItem = e.target.parentNode.parentNode.parentNode;
+      const contentList = document.querySelector(".product-list");
+
+      /*<div class="selected-item-quantity">
```
Comment (smtrd3) [inline script.js]: no comments

File: index.html [hunk 9]
```diff
@@ -6,300 +6,374 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
```
Comment (smtrd3) [inline index.html]: if we are not using the file then you can remove this file for this branch. I believe in react version we don't need this html file.

File: my-app-react/README.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
+# Getting Started with Create React App
```
Comment (smtrd3) [inline my-app-react/README.md]: this file is not needed, lets delete this

File: my-app-react/public/index.html [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta
+      name="description"
+      content="Web site created using create-react-app"
+    />
+    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
+    <!--
```
Comment (smtrd3) [inline my-app-react/public/index.html]: remove this comments.

File: my-app-react/src/App.js [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+import menuList from "./Models/menuListModel";
+const App = () => {
+  const [activeMenu, setActiveMenu] = useState("Recommended");
+  const [activeMenuItems, setActiveMenuItem] = useState(menuItems[activeMenu]);
+  const [currCartItems, setCartItems] = useState([]);
+  function selectedHandler(currItem) {
+    setActiveMenu(currItem);
+    let updatedMenuItems = [];
+    updatedMenuItems = [...updatedMenuItems, ...menuItems[currItem]]
+    console.log("update after selection :",updatedMenuItems)
```
Comment (smtrd3) [inline my-app-react/src/App.js]: remove console.logs

File: my-app-react/src/App.js [hunk 13]
```diff
@@ -0,0 +1,168 @@
...
+    setActiveMenuItem(updatedMenuItems);
+  }
+  function increaseInCart(myname) {
+    console.log("increase in cart", myname);
```
Comment (smtrd3) [inline my-app-react/src/App.js]: remove console.lgs

File: my-app-react/src/App.js [hunk 14]
```diff
@@ -0,0 +1,168 @@
...
+    let updatedCartItems;
+    console.log("before loop : ", currCartItems);
+    for (let i = 0; i < currCartItems.length; i++) {
```
Comment (smtrd3) [inline my-app-react/src/App.js]: no for loops

File: my-app-react/src/App.js [hunk 15]
```diff
@@ -0,0 +1,168 @@
...
+        console.log("updated one : ", updatedCartItems);
+        console.log("old one : ", currCartItems);
+        break;
+      }
+    }
+    setCartItems(updatedCartItems);
+  }
+  function addToCart(newCartItem) {
+    const updatedCartItems = [...currCartItems, newCartItem];
+    console.log("new cart items", updatedCartItems);
```
Comment (smtrd3) [inline my-app-react/src/App.js]: remove console.lgs

File: my-app-react/src/App.js [hunk 16]
```diff
@@ -0,0 +1,168 @@
...
+    setCartItems(updatedCartItems);
+  }
+  function onSearchDish(keyword) {
+    console.log(keyword);
```
Comment (smtrd3) [inline my-app-react/src/App.js]: remove console.lgs

File: my-app-react/src/App.js [hunk 17]
```diff
@@ -0,0 +1,168 @@
...
+    if (keyword === "") {
+      setActiveMenu(activeMenu);
+    }
+    let updatedMenuItems = [];
+    for (let i = 0; i < menuList.length; i++) {
```
Comment (smtrd3) [inline my-app-react/src/App.js]: no for loops

File: my-app-react/src/App.js [hunk 18]
```diff
@@ -0,0 +1,168 @@
...
+        curr.dishName.toLowerCase().includes(keyword.toLowerCase())
+      );
+      updatedMenuItems = [...updatedMenuItems, ...searchInMenu];
+    }
+    setActiveMenuItem(updatedMenuItems);
+  }
+  function onFilter(isChecked) {
+    let updatedMenuItems = [];
+    if (isChecked) {
+      for (let i = 0; i < menuList.length; i++) {
```
Comment (smtrd3) [inline my-app-react/src/App.js]: no for loops

File: my-app-react/src/App.js [hunk 19]
```diff
@@ -0,0 +1,168 @@
...
+    else {
+      updatedMenuItems = menuItems[activeMenu];
+    }
+    console.log(updatedMenuItems);
+    setActiveMenuItem(updatedMenuItems);
+    
+  }
+  function handlePromise() {
+    return new Promise(function (resolve, reject) {
+      const userdata = fetch("https://api.github.com/users").then(
```
Comment (smtrd3) [inline my-app-react/src/App.js]: where are we using this api

File: my-app-react/src/App.js [hunk 20]
```diff
@@ -0,0 +1,168 @@
...
+          }
+        }
+      );
+    });
+  }
+  function addToLocalStorage() {
+    handlePromise().then(
+      function () {
+        localStorage.setItem("cartList", JSON.stringify(currCartItems));
+        console.log(JSON.parse(localStorage.getItem("cartList")));
```
Comment (smtrd3) [inline my-app-react/src/App.js]: no console.log

File: my-app-react/src/App.js [hunk 21]
```diff
@@ -0,0 +1,168 @@
...
+      },
+      function (param) {
```
Comment (smtrd3) [inline my-app-react/src/App.js]: what is this function doing here?

File: my-app-react/src/App.js [hunk 22]
```diff
@@ -0,0 +1,168 @@
...
+      }
+    );
+  }
+
+  useEffect(() => {
+    setActiveMenu(activeMenu);
+    setActiveMenuItem(activeMenuItems);
+    setCartItems(currCartItems);
+    console.log("after veg filter", currCartItems);
+    console.log("my menu items", activeMenuItems);
```
Comment (smtrd3) [inline my-app-react/src/App.js]: no console.log

File: my-app-react/src/Components/CartItems/CartItems.js [hunk 23]
```diff
@@ -0,0 +1,79 @@
+import "./cartItems.style.css";
+import { VegLogo, NonVegLogo } from "../../Models/ImageConstants";
+function CartItems(props) {
+  console.log("Re render :", props.allCartItems.length);
```
Comment (smtrd3) [inline my-app-react/src/Components/CartItems/CartItems.js]: no console.log

File: my-app-react/src/Components/MenuContentList/MenuContentList.js [hunk 24]
```diff
@@ -0,0 +1,88 @@
...
+    <div className="menu-content-list">
+      <div className="menuHeading">
+        <h1 className="content-List-heading">{props.activeMenu}</h1>
+        <h3 className="num-items">{props.activeMenuList.length} Items</h3>
+      </div>
+      <ul className="product-list">
+        {props.activeMenuList.map((curr) => {
+          const { dishName, isVeg, price, desc, image } = curr;
+          let qty = 0;
+          for (let i = 0; i < props.currCartItems.length; i++) {
```
Comment (smtrd3) [inline my-app-react/src/Components/MenuContentList/MenuContentList.js:25]: no for loops

File: style.css [hunk 25]
```diff
@@ -1,44 +1,45 @@
 @import url("https://fonts.googleapis.com/css2?family=Lato:wght@300;400&display=swap");
```
Comment (smtrd3) [inline style.css]: do we need this file?

File: script.js [hunk 26]
```diff
@@ -0,0 +1,429 @@
+import menuItems from "./Models/DishModel.js";
+import { NonVegLogo, VegLogo } from "./Models/ImageConstants.js";
```
Comment (smtrd3) [inline script.js]: do we need this file? I guess its from MVC example, so in react task you should remove this.

**Review by smtrd3 — CHANGES_REQUESTED** [review]
Things to take care immediately 

- no unnecessary comments or console.logs
- no for loops
- keep only those files which are actually used, please delete the files from the MVC task 
- are there duplicate images? If yes please remove duplicates.
- code formatting (very urgent)

File: style.css (hunk 25 above)
Comment (deepika845) [reply style.css]: ya to provide font style as suggested in excel sheet

File: my-app-react/src/App.js (hunk 1 above)
Comment (deepika845) [reply my-app-react/src/App.js]: removed console logs

File: my-app-react/src/App.js (hunk 2 above)
Comment (deepika845) [reply my-app-react/src/App.js:118]: formatted using prettier

File: my-app-react/src/Components/MenuList/MenuList.js (hunk 3 above)
Comment (deepika845) [reply my-app-react/src/Components/MenuList/MenuList.js]: removed comments

File: script.js (hunk 4 above)
Comment (deepika845) [reply script.js]: it was unused file, removed that

File: my-app-react/src/App.js (hunk 22 above)
Comment (deepika845) [reply my-app-react/src/App.js]: removed

File: my-app-react/src/Components/CartItems/CartItems.js (hunk 23 above)
Comment (deepika845) [reply my-app-react/src/Components/CartItems/CartItems.js]: removed

File: script.js (hunk 26 above)
Comment (deepika845) [reply script.js]: removed

File: my-app-react/src/Components/MenuContentList/MenuContentList.js (hunk 24 above)
Comment (deepika845) [reply my-app-react/src/Components/MenuContentList/MenuContentList.js:25]: removed

**Pull request closed without merging**